components/tools/OmeroM/src/annotations/writeMapAnnotation.m

```diff
@@ -46,16 +46,19 @@
 % Input check
 ip = inputParser;
 ip.addRequired('session');
-ip.addRequired('keys', @(x) ischar(x) || iscellstr(x));
-ip.addRequired('values', @(x) ischar(x) || iscellstr(x));
-ip.addParamValue('namespace', '', @ischar);
-ip.addParamValue('description', '', @ischar);
-ip.addParamValue('group', [], @(x) isscalar(x) && isnumeric(x));
+ip.addRequired('keys', @(x) ischar(x) || iscellstr(x) || isstring(x));
+ip.addRequired('values', @(x) ischar(x) || iscellstr(x) || isstring(x));
+ip.addParameter('namespace', '', @ischar);
+ip.addParameter('description', '', @ischar);
+ip.addParameter('group', [], @(x) isscalar(x) && isnumeric(x));
 ip.parse(session, keys, values, varargin{:});
 
 % Convert keys and values into cell arrays
 if ischar(keys), keys = {keys}; end
 if ischar(values), values = {values}; end
+if isstring(keys), keys = cellstr(keys); end
+if isstring(values),values = cellstr(values); end
```

Comment:
This seems to cause an error in our matlab training examples:
`Undefined function 'isstring' for input arguments of type 'cell'.`
https://ci.openmicroscopy.org/view/Failing/job/OMERO-DEV-merge-training/995/console

I guess that happens somewhere around here: https://github.com/openmicroscopy/openmicroscopy/blob/c60092a554f8295c2992c407a94a2623011306b5/examples/Training/matlab/WriteData.m#L426 

Unfortunately I'm not familiar enough with the matlab code to do some more detailed debugging.

Reply:
`isstring` is relatively new in MATLAB. I can make it backward compatible. Hold on for 2 weeks.

Reply:
We are also running the training example with an old version of Matlab (2013b)
We should discuss what will be a reasonable minimum version to expect

Reply:
A lot of changes have been introduced in R2014b. Essentially, .....

```matlab
if verLessThan('matlab','8.4.0')
     % execute code for R2014a or earlier
else
     % execute code for R2014b or later
end
```